lib/model/matches.ts

```diff
@@ -56,22 +61,26 @@ export async function getMinMaxMatchTimestamps(gamer){
 export async function getMatchDetailsFromAPI(queryTimeframes, gamer, api, sleepTime = MATCH_DETAILS_SLEEP_TIME) {
     console.log('# of Details API calls needed for gamer:' + gamer.username + ' on platform:' + gamer.platform + ': ' + queryTimeframes.length);
     console.log('getting match details for: ' + gamer.username + ' on platform:' + gamer.platform);
+
     const matches = await Bluebird.mapSeries(queryTimeframes, async (item) => {
         const output = await api.MWcombatwzdate(gamer.username, item.start, item.end, gamer.platform);
-        const matches =  output.matches || [];
+        const matches = UtilityService.validateItem(output.matches, 'array', []);
+
         console.log('Num matches for interval for gamer:' + gamer.username + ' on platform:' + gamer.platform + ':' + matches.length);
+
         UtilityService.sleep(sleepTime);
+
         return matches;
     });
-    return matches.flatMap((matchArr)=> matchArr);
-}
 
+    return UtilityService.validateItem(matches, 'array', []).flatMap((matchArr) => matchArr);
```

Comment:
Thought something was breaking in here, but I realized it was that your node version was different than mine (10, which didn't support the flatMap). Nothing breaks or changes here, I did test this in node v12

Reply:
Which also reminds me that we should set some rules to explicitly require that you run the exact version of node each time.